File: tlg/src/main/java/ch/bailu/tlg/StateHighscore.java
package ch.bailu.tlg;

import java.io.IOException;

public class StateHighscore extends State {
    public static final int ID=1;

    public StateHighscore(InternalContext c) {
        super(c);
    }

    @Override
    public State init(PlatformContext c) {
        context.state=this;
        c.onNewHighscore();
        return this;
    }

    @Override
    public State setHighscoreName(PlatformContext c, String name) {
        if (name.length()>0) {
            HighscoreList highscoreList = new HighscoreList(c);

            highscoreList.add(name, context.currentScore.getScore());
            try {
                highscoreList.writeState(c);
            } catch (IOException e) {
                e.printStackTrace();
            }

            return new StateLocked(context).init(c);

        } else {
            return this;
        }
    }

    @Override
    public int getID() {
        return ID;
    }

    @Override
    public String toString() {
        return "New Highscore!";
    }
}

File: tlg/src/main/kotlin/tlg/state/StateHighscore.kt
package tlg.state


import tlg.context.InternalContext
import tlg.context.PlatformContext
import tlg.score.HighScoreList
import java.io.IOException

class StateHighscore(c: InternalContext) : State(c) {
    override fun init(c: PlatformContext): State {
        context.state = this
        c.onNewHighscore()
        return this
    }

    override fun setHighscoreName(c: PlatformContext, name: String): State {
        return if (name.isNotEmpty()) {
            val highscoreList = HighScoreList(c)
            highscoreList.add(name, context.currentScore.score)
            try {
                highscoreList.writeState(c)
            } catch (e: IOException) {
                e.printStackTrace()
            }
            StateLocked(context).init(c)
        } else {
            this
        }
    }

    override fun toString(): String {
        return "New Highscore!"
    }

    override val id = ID

    companion object {
        const val ID = 5
    }
}
